client/src/api/client.ts

```diff
@@ -0,0 +1,540 @@
+import { errors } from '../util/errors';
+import type * as Types from './components';
+import { operationsByTag } from './components';
+import type { FetcherExtraProps, FetchImpl } from './fetcher';
+import { ApiProviders, getApiUrl } from './providers';
+import type * as Responses from './responses';
+import type * as Schemas from './schemas';
+
+export interface XataApiClientOptions {
+  fetchImpl: FetchImpl;
+  apiKey: string;
+  api?: ApiProviders;
```

Comment:
I think we could better use a `hostname`. People are not going to use it now, but may be confusing if they see `production` somewhere in our code and they can think this is about their production environment or something like that, or if they think `api` has something to do with API versioning.

And potentially in the future if we provide a local development server or if we provide on-prem xata instances, it makes more sense to have the hostname fully customizable. WDYT?

I know we have two different hosts in production (api.xata.io and xata.sh) but that's the exception. So I'd only add one hostname parameter, and if not specified, use the specific default one for each case (api.xata.io or xata.sh).

Reply:
Makes sense